rfcs/rfc-colin-states-mapping-to-lear.md

```diff
@@ -0,0 +1,154 @@
+- Start Date: 2021-12-03
+- Target Major Version: 
+- Reference Issues: [10098](https://github.com/bcgov/entity/issues/4797)
+- Entity Issue: [8285](https://github.com/bcgov/entity/issues/14691)
+- Implementation PR: (leave this empty)
+
+# Summary
+
+COLIN represents a business state as two states which are referred to as an operational state(`CORP_OP_STATE` table) and a corp state(`CORP_STATE`) table.  In many instances, these two states are used in conjunction to drive business logic and to determine allowable business state transitions between `ACTIVE` and `HISTORICAL` states for a given business.
+
+LEAR will not be adopting the COLIN approach of using the `CORP_STATE` table.  
+
+This RFC will provide design details as to how LEAR will achieve the same functionality as COLIN without the use of the `CORP_STATE` table.
+
+Note that the [Business States RFC](rfc-business-states.md) as well the current implementation of the Legal API was used as a starting point for the creation of this document.
+
+#### COLIN Database Structure for Business States
+![colin_states.png](rfc-colin-states-mapping-to-lear/colin_states.png)
+
+#### Current Corp State Data in CPRD
+_`ACT/HIS` column is `CORP_OP_STATE` table._
+_`Type` column is `CORP_STATE` table._
+![colin_corp_state_stats.png](rfc-colin-states-mapping-to-lear/colin_corp_state_stats.png)
+
+
+# Basic design
+
+The COLIN operational states can be mapped 1-1 to the business states(`businesses.state`) in LEAR currently.  Only `ACTIVE` and `HISTORICAL` states are used by COLIN.  Additional states may be added to LEAR if it is determined to be required based on analysis and work done when implementing filings such as _liquidation_ and _involuntary dissolution_.
+
+The COLIN `CORP_STATE` table can be represented in LEAR by using a combination of the filing type, dynamic calculations (warning structures or business properties like good standing) or using the filing type that transitioned a business into its current state.
+
+Allowable COLIN business state transitions between `HISTORICAL` and `ACTIVE` will be handled using a combination of the filing that transitioned the business to its current state and a map that provides allowable business state transitions.
```

Comment:
We need to change this language a little, as _state_ is a confusing and overloaded term.
There are 2 legal states for a business **active** and **historical**
For now _Compliance_ is informational only, it doesn't block any registry actions but must be noted to viewers of the Registry.
_Good standing_ has various statuses and can lead to a state transition
_Liquidation_ has a number of _compliance_ affecting things and blocks most filings, it also has various statuses and eventually ends with the company being made **historical** or can be **active** if not yet struck from the Registry. Afaik there is no legal way to restore a company once it's liquidation has been completed.

Reply:
Updated basic design section to hopefully make thing less confusing.  Also included some points mentioned in previous(^^) comment.